### ISSUE TITLE
Time dependent and time zero depth calculations

### ISSUE THREAD

**Issue by anwiseNOCL** [British-Oceanographic-Data-Centre/COAsT#109]
**Problem**
The user should be able to access the depth at a specific grid point on each gird. For example when plotting a vertical temperature profile, the user may want to plot with a depth axis. Temperature is defined on T points and so there must be a depth defined on the T points. The same will apply to a vertical velocity profile on the U and V grids.

**Objective**
1. Create a depth variable in the NEMO class xarray dataset that will hold the 4d depth fields. 

2. Modify the 'depth at time zero' variable in the NEMO class xarray dataset to include depth_0 on the u and v grids.

**A/C**
 depth (time_dim,z_dim,y_dim,x_dim) and depth_0 (z_dim,y_dim,x_dim) variables in the NEMO class xarray dataset that will define the depth on their respective grids.

**Comment by jpolton** [comment]
@anwiseNOCL currently depth_0 is a `coordinate` for the NEMO DataSet, as it is established at the start of things. Would depth also be a replacement (or additional?) coordinate, or just a regular variable? I suspect that there are Xarray advantages / problems associated with this.

**Comment by anwiseNOCL** [comment]
@jpolton I imagine a time dependent depth would replace depth_0 as the coordinate and depth_0 would become a regular variable. The only problem i see with that is that it time dependent depth is much heavier than depth_0. Not sure how urgent this is as I don't think the levels change that much do they? and it is only for plotting.

I have now coded up the depth_0 calculation for u, v and f so this is all fine now. Just trying to get it to play nice with my transect stuff then will do pull request to merge it into 104. Hopefully today.

**Comment by jpolton** [comment]
@anwiseNOCL Agreed the the time_dependent depths is not urgent. I've added a tag:someday in anticipation of us wrapping up the 104 branch (incl. the t_0 stuff) in the next week. Leaving just the time dependant stuff on this ticket.

My pull request might slightly change how depth_0 is incorporated for u,v,f-grids. I'd be happy to have a go if it is easier?

**Comment by anwiseNOCL** [comment]
@jpolton I have already implemented the depth_0 calculation. It does it on creation of the NEMO object and assigns to depth_0 for the specific grid. I'm not sure there is a different way to incorporate this?
